client/src/app/common/CustomRules/rules-utils.tsx

```diff
@@ -11,12 +11,13 @@ export const checkRuleFileType = (filename: string): RuleFileType => {
     return "XML";
   } else return null;
 };
-
+type ParsedYaml = any[] | {};
 export const parseRules = (file: IReadFile): ParsedRule => {
   if (file.data) {
     if (checkRuleFileType(file.fileName) === "YAML") {
-      const yamlDoc = yaml.load(file.data) as any[];
-      const yamlLabels = yamlDoc?.reduce((acc, parsedLine) => {
+      const yamlDoc: ParsedYaml = yaml.load(file.data) as ParsedYaml;
+      const yamlList = Array.isArray(yamlDoc) ? yamlDoc : [yamlDoc];
+      const yamlLabels = yamlList?.reduce((acc, parsedLine) => {
```

Comment:
Also (and it's fine if we leave this because the `reduce` was pre-existing) if your reduce is just adding elements to an accumulated array, you can simplify it by using [flatMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) instead. Each iteration of a flatMap returns an array, and the arrays are all flattened/appended together at the end.

```ts
const yamlLabels = yamlList?.flatMap((parsedLine) => {
  return parsedLine?.labels ? parsedLine?.labels : [];
}) || [];
```

Note that this is the same as `.map(...).flat(1)`.

Also, looking at the behavior of `getLabels`, if the same label appears more than once in the YAML it's going to end up duplicated in `otherLabels` and `allLabels`, right? I wonder if that's a problem. If it is, we could deduplicate the array by using a Set.

```ts
const yamlLabels = Array.from(new Set(yamlList?.flatMap((parsedLine) => {
  return parsedLine?.labels ? parsedLine?.labels : [];
}) || []);
```